docs/upgrade-guide.md

```diff
@@ -2,12 +2,45 @@
 
 ## Upgrading from deck.gl v4 to v4.1
 
-A highly compatible release, the biggest change is that deck.gl v4.1 brings in luma.gl v4. It is a highly compatible release but there are some changes. If you are writing deck.gl layers using luma.gl classes, please refer to the Upgrade Guide for luma.gl.
+deck.gl v4.1 is a backward-compatible release. Most of the functionality and APIs remain unchanged but there are smaller changes that might requires developers' attention if they /**develop their own layers**.
```

Comment:
backward compatible release with v4.0 (maybe redundant for those who's familiar with semver)